src/github.rs

```diff
@@ -1499,6 +1499,18 @@ impl Repository {
         }
     }
 
+    pub async fn github_commit(
+        &self,
+        client: &GithubClient,
+        sha: &str,
+    ) -> anyhow::Result<GithubCommit> {
+        let url = format!("{}/commits/{}", self.url(client), sha);
+        client
+            .json(client.get(&url))
+            .await
+            .with_context(|| format!("{} failed to get github commit {sha}", self.full_name))
+    }
```

Comment:
nit: while `Repository::git_commit` only returns a `GitCommit` it has a `author.date` field, making this addition a bit useless for the purpose of this handler, but this seems like a good addition anyway